packages/flutter_tools/lib/src/commands/packages.dart

```diff
@@ -357,3 +349,60 @@ class PackagesGetCommand extends FlutterCommand {
     );
   }
 }
+
+// TODO: this doesn't seem right
+Pub get pub => context.get<Pub>()!;
+
+class PackagesPostGetHook extends FlutterCommand {
+  PackagesPostGetHook(this._context) {
+    argParser.addOption('directory', abbr: 'C', mandatory: true);
+    argParser.addFlag('example', defaultsTo: true);
+    argParser.addFlag('update-version-and-package-config');
+    argParser.addFlag('regenerate-platform-specific-tooling');
+  }
+  final PubContext _context;
+
+  @override
+  bool get hidden => true;
+
+  @override
+  // TODO: implement description
+  String get description => throw UnimplementedError();
+
+  @override
+  // TODO: implement name
+  String get name => '_post_pub_get';
+
+  @override
+  Future<FlutterCommandResult> runCommand() async {
+    // TODO: think about this part that actually happens before _runStdio (pre-get-hook?)
+    // - generateLocalizationsSyntheticPackage
+
+    final String directoryOption = argResults!['directory'] as String;
+    final String? target = findProjectRoot(globals.fs, directoryOption);
+    if (target == null) {
+      throwToolExit('Expected to find project root in $directoryOption.');
+    }
+    final FlutterProject rootProject = FlutterProject.fromDirectory(globals.fs.directory(target));
+
+    final bool example = argResults!['example'] as bool;
+    final bool updateVersionAndPackageConfig = argResults!['update-version-and-package-config'] as bool;
```

Comment:
What is the logic that determines what value this (and the next one) should be?

Reply:
These are passed unconditionally by the dart pub tool. I added them to help with testing and for flexibility, but from what @sigurdm said, it seems like they are probably going to cause issues in the future, so I'll remove them. https://github.com/dart-lang/pub/pull/3745#discussion_r1088735379

Reply:
SGTM.